sc-memory/sc-memory/scs/scs.g4

```diff
@@ -222,41 +222,67 @@ idtf_edge returns [ElementHandle handle]
   ;
   
 idtf_set returns [ElementHandle handle]
-  : '{' 
-      { 
-        std::string const setIdtf = "..set_" + std::to_string($ctx->start->getLine()) + "_" + std::to_string($ctx->start->getCharPositionInLine());
-        $ctx->handle = m_parser->ProcessIdentifier(setIdtf);
-        ElementHandle const typeEdge = m_parser->ProcessConnector("->");
-        ElementHandle const typeClass = m_parser->ProcessIdentifier("sc_node_tuple");
-
-        m_parser->ProcessTriple(typeClass, typeEdge, $ctx->handle);
+  : '{' ctx_s=idtf_set_elements["set"] { $ctx->handle = $ctx->ctx_s->handle; } '}'
+  | '<' ctx_v=idtf_set_elements["vector"] { $ctx->handle = $ctx->ctx_v->handle; } '>'
+  ;
+
+idtf_set_elements [std::string setType]
+  returns [ElementHandle handle]
+  locals [ElementHandle prevEdge]
+  : {
+      std::string const setIdtf = "..set_" + std::to_string($ctx->start->getLine()) + "_" + std::to_string($ctx->start->getCharPositionInLine());
+      $ctx->handle = m_parser->ProcessIdentifier(setIdtf);
+      ElementHandle const typeEdge = m_parser->ProcessConnector("->");
+      ElementHandle const typeClass = m_parser->ProcessIdentifier("sc_node_tuple");
+
+      m_parser->ProcessTriple(typeClass, typeEdge, $ctx->handle);
+    }
+  a1=attr_list? i1=idtf_common
+    {
+      ElementHandle const edge = m_parser->ProcessConnector("->");
+      m_parser->ProcessTriple($ctx->handle, edge, $ctx->i1->handle);
+
+      if ($ctx->a1 != nullptr)
+      {
+        APPEND_ATTRS($ctx->a1->items, edge);
```

Comment:
can we make naming more clear? Or such refactor should be in another PR?

Reply:
https://github.com/ostis-ai/sc-machine/issues/196